media-proxy/src/mesh/conn_rdma.cc

```diff
@@ -0,0 +1,359 @@
+                       const std::string& dev_port, libfabric_ctx *& dev_handle)
+{
+
+    trx_sz = request.payload_args.rdma_args.transfer_size;
+    _kind = kind();
+
+    std::memset(&ep_cfg, 0, sizeof(ep_cfg));
+    std::memcpy(&ep_cfg.remote_addr, &request.remote_addr, sizeof(request.remote_addr));
+    std::memcpy(&ep_cfg.local_addr, &request.local_addr, sizeof(request.local_addr));
+    
+    ep_cfg.dir = _kind == Kind::receiver ? direction::RX : direction::TX;
+
+    mDevHandle = dev_handle;
+
+    queue_size = request.payload_args.rdma_args.queue_size;
+
+    set_state(ctx, State::configured);
+    return Result::success;
+}
+
+/**
+ * @brief Establishes an RDMA connection by initializing the device, endpoint, buffer queue,
+ * and starting necessary threads for processing RDMA operations.
+ *
+ * Ensures proper cleanup and state updates on failure. Returns success if all steps
+ * (device initialization, endpoint setup, buffer queue creation, and endpoint configuration)
+ * complete successfully.
+ *
+ * @param ctx The operation context.
+ * @return Result::success on success, or an appropriate error result on failure.
+ */
+Result Rdma::on_establish(context::Context& ctx)
+{
+    if (init) {
+        log::error("RDMA device is already initialized")("state", "initialized");
+        set_state(ctx, State::active);
+        return Result::error_already_initialized;
+    }
+
+    init_buf_available();
+
+    int ret;
+    Result res;
+
+    if (!mDevHandle) {
+        ret = libfabric_dev_ops.rdma_init(&mDevHandle);
+        if (ret) {
+            log::error("Failed to initialize RDMA device")("ret", ret);
+            set_state(ctx, State::closed);
+            return Result::error_initialization_failed;
+        }
+    }
+
+    ep_cfg.rdma_ctx = mDevHandle;
+
+    ret = libfabric_ep_ops.ep_init(&ep_ctx, &ep_cfg);
+    if (ret) {
+        log::error("Failed to initialize RDMA endpoint context")("error", fi_strerror(-ret));
+        set_state(ctx, State::closed);
+        return Result::error_initialization_failed;
+    }
+    if(queue_size == 0) {
+        log::error("Queue size is not set")("queue_size", queue_size);
+        set_state(ctx, State::closed);
+        return Result::error_bad_argument;
+    }
+
+    res = init_queue_with_elements(queue_size, trx_sz);
+    if (res != Result::success) {
+        log::error("Failed to initialize buffer queue")("trx_sz", trx_sz);
+        if (ep_ctx) {
+            libfabric_ep_ops.ep_destroy(&ep_ctx);
+        }
+        set_state(ctx, State::closed);
+        return res;
+    }
+
+    // Configure RDMA endpoint
+    res = configure_endpoint(ctx);
+    if (res != Result::success) {
+        log::error("RDMA configuring failed")("state", "closed");
+        if (ep_ctx) {
+            libfabric_ep_ops.ep_destroy(&ep_ctx);
+        }
+        set_state(ctx, State::closed);
+        return res;
+    }
+
+    init = true;
+    res = start_threads(ctx);
```

Comment:
res is never checked

Reply:
fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/286/commits/390a65332407a21263d7269b722310576780ad64